### PR TITLE
Adding a polyfill files for IE

### PR DESCRIPTION
**Pull request by fforres**
First of all, nice project :) 
I was thinking on making something like this to track usage of the predictions and actual user behaviour in a couple of my projects. 
Kudos @mathisonian  :) 

----- 

PR is regarding [this issue. #2](https://github.com/mathisonian/premonish/issues/2) .

- [x] Created and importing Polyfills file.
- [x] Run `npm run example` and `npm run syntax-fix` before creating the pr.  (It does some syntax fixes some on `examples/debug.js` and `src/index.js` which weren't present before)

Also, I used yarn, and it created a `.lock` file let me know if maybe you folks want it removed.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -163,6 +165,4 @@ var Premonish = function () {
   return Premonish;
 }();
 
-;
-
-exports.default = Premonish;
+exports.default = Premonish;
```
Comment (Qix-) [inline lib/index.js:168]: Please keep the new line.

File: lib/polyfills.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  (function () {
+    Math.sign = function (x) {
+      x = +x; // convert to a number
+      if (x === 0 || isNaN(x)) {
+        return Number(x);
+      }
+      return x > 0 ? 1 : -1;
+    };
+  });
+}
```
Comment (Qix-) [inline lib/polyfills.js]: Please keep the new line.

File: src/index.js [hunk 3]
```diff
@@ -109,23 +110,23 @@ class Premonish {
...
     this._onMouseMove = callback;
   }
 
-  onIntent(callback) {
+  onIntent (callback) {
     this._onIntent = callback;
   }
 
-  stop() {
+  stop () {
```
Comment (Qix-) [inline src/index.js:126]: Why all these added spaces?

File: src/polyfills.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  (function () {
+    Math.sign = function (x) {
+      x = +x; // convert to a number
+      if (x === 0 || isNaN(x)) {
+        return Number(x);
+      }
+      return x > 0 ? 1 : -1;
+    };
+  });
+}
```
Comment (Qix-) [inline src/polyfills.js]: Please format accordingly.

**Review by Qix- — CHANGES_REQUESTED** [review]
The lock file should be .gitignored (but preferably in a separate PR).

Also, try to keep individual PRs free of any unrelated format changes :) it makes things harder to revert if need be since it introduces unrelated changes.

**Comment by mathisonian** [comment]
Thanks for the PR @fforres! 

Could you do 2 things and then I'll merge? 

1. Instead of polyfilling `Math.sign` could you just make a `sign` function (see https://github.com/sindresorhus/ponyfill for the reasoning behind this)
2. Add yarn lock to gitignore

File: src/index.js (hunk 3 above)
Comment (fforres) [reply src/index.js:126]: These are from running `npm run syntax-fix`.

**Comment by fforres** [comment]
PR updated by the comments on @mathisonian.

But @Qix- points are valid.
I run `npm run syntax-fix` maybe thinking it was there as a `contributing guideline` of sorts. 
Should I revert those syntax changes it created?

**Comment by mathisonian** [comment]
Thanks! 

@fforres @Qix-  apologies for the confusion about coding style, I put this together pretty quickly and forgot to run my own linter, so thats why you are seeing the whitespace differences.

File: src/index.js (hunk 3 above)
Comment (Qix-) [reply src/index.js:126]: Gotcha :)

**Pull request merged**